pkg/querier/frontend/record_rule_substitution.go

```diff
@@ -0,0 +1,331 @@
+package frontend
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/weaveworks/common/user"
+	"gopkg.in/fsnotify/fsnotify.v1"
```

Comment:
I'm not a big fan of fsnotify, as it can miss events etc.  Can we just reload the config every minute or so?  

Also, can you attempt to unify / reuse the code in util/validation/overrides.go, perhaps pulling it out into util/reloader.go?